File: tools/serve.py
#!/usr/bin/env python3
"""Local static server with the same two NFL feed rewrites as Vercel."""
import argparse
import pathlib
import urllib.error
import urllib.request
from http.server import SimpleHTTPRequestHandler, ThreadingHTTPServer
from urllib.parse import urlsplit

ROOT = pathlib.Path(__file__).resolve().parents[1]
FEEDS = {"/feeds/nfl/injuries": "injuries", "/feeds/nfl/scoreboard": "scoreboard"}

class Handler(SimpleHTTPRequestHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, directory=str(ROOT), **kwargs)

    def do_GET(self):
        url = urlsplit(self.path)
        if url.path not in FEEDS:
            return super().do_GET()
        upstream = "https://site.api.espn.com/apis/site/v2/sports/football/nfl/" + FEEDS[url.path]
        if url.query:
            upstream += "?" + url.query
        try:
            with urllib.request.urlopen(upstream, timeout=12) as response:
                body = response.read()
            self.send_response(200)
            self.send_header("Content-Type", "application/json")
            self.send_header("Cache-Control", "no-store")
            self.end_headers()
            self.wfile.write(body)
        except (urllib.error.URLError, TimeoutError):
            self.send_error(503, "NFL feed unavailable")

if __name__ == "__main__":
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("--port", type=int, default=8767)
    args = parser.parse_args()
    ThreadingHTTPServer(("127.0.0.1", args.port), Handler).serve_forever()

File: tools/serve.mjs
// Local equivalent of the static Vercel site, its fixed feed rewrites, and projection function.
import {createServer} from 'node:http';
import {readFile} from 'node:fs/promises';
import {resolve,extname} from 'node:path';
import {fileURLToPath} from 'node:url';
import {createRequire} from 'node:module';
const root=fileURLToPath(new URL('..',import.meta.url));
const require=createRequire(import.meta.url);
const projections=require('../api/projections.js');
const handlers=Object.fromEntries(['party-parlays','campaign','edge-record','season-alerts','season-scan'].map(n=>['/api/'+n,async(req,res)=>require('../api/'+n+'.js')(req,res)]));
const mime={'.html':'text/html','.js':'text/javascript','.css':'text/css','.json':'application/json','.svg':'image/svg+xml','.png':'image/png','.woff2':'font/woff2'};
const feeds={'/feeds/nfl/injuries':'injuries','/feeds/nfl/scoreboard':'scoreboard'};
const port=Number(process.argv[process.argv.indexOf('--port')+1])||8767;
createServer(async(req,res)=>{
  try{
    const url=new URL(req.url,'http://localhost');
    if(handlers[url.pathname]) return await handlers[url.pathname](req,res);
    if(url.pathname==='/api/projections') return await projections(req,res);
    if(feeds[url.pathname]){
      const upstream=await fetch('https://site.api.espn.com/apis/site/v2/sports/football/nfl/'+feeds[url.pathname]+url.search,{signal:AbortSignal.timeout(18000)});
      res.writeHead(upstream.status,{'Content-Type':'application/json','Cache-Control':'no-store'});
      res.end(Buffer.from(await upstream.arrayBuffer()));return;
    }
    const path=['/','/draft','/season'].includes(url.pathname)?'/index.html':url.pathname==='/bets'?'/bets.html':decodeURIComponent(url.pathname);
    const file=resolve(root,'.'+path);
    if(!file.startsWith(root)){res.writeHead(403);res.end();return;}
    const body=await readFile(file);res.writeHead(200,{'Content-Type':mime[extname(file)]||'application/octet-stream'});res.end(body);
  }catch(e){res.writeHead(e.code==='ENOENT'?404:502);res.end('Resource unavailable');}
}).listen(port,'127.0.0.1',()=>console.log('War Room: http://127.0.0.1:'+port));
